packages/cozy-harvest-lib/src/components/hooks/useMaintenanceStatus.jsx

```diff
@@ -1,6 +1,14 @@
 import { useState, useEffect } from 'react'
 import { Registry } from 'cozy-client'
 import get from 'lodash/get'
+import memoize from 'lodash/memoize'
+
+const memoizedFetchApp = memoize(
+  (registry, slug) => {
+    return registry.fetchApp(slug)
+  },
+  (registry, slug) => slug
+)
```

Comment:
I understand that you did that because this fetch can be slow (is it correct?) but having it memoized forever makes me inconfortable. I know that this is something we don't change often, but... 

I think the fix should rather be on the registry side of cozy-client so that it uses the redux store to persist data. So we would make sure to read the store first?

Reply:
It's memoized until the user reloads the page, to have this functionality in cozy-client would be a much bigger endeavour. If you are unconfortable with this, I prefer to remove this commit.

Reply:
Yeah this is what I meant when using "forever" (until the user reloads the page). 

I'll create an issue on Cozy-Client to think about changing the Registry behavior. In the meantime, I just rise this potential issue, you deal with konnectors more often than me, so if you decide that we can memoize the result since we don't change the konnector's status pretty often, let's keep it 👍